scripts/split.sh

```diff
@@ -99,26 +103,38 @@ process() {
 
     if [ $hash_index -gt $TRAINPCT ]
     then
+      let "n_test++"
       target=$TESTDIR/$file
       echo -n "T"
     else
+      let "n_train++"
       target=$TRAINDIR/$file
       echo -n "*"
     fi
 
     ln $dir/$file $target
 
-    # exceeding max buffer size, remove overhead
-    if [ $n -gt $max ]
+    # exceeding max buffer size for either, lock and remove overhead as appropriate
+    if [ $n_test -gt $max_test ] || [ $n_train -gt $max_train ]
     then
       (
       flock -e 200
-      ls -rt $TESTDIR | head -n $overhead_test | xargs -I{} rm -f $TESTDIR/{}
-      ls -rt $TRAINDIR | head -n $overhead_train | xargs -I{} rm -f $TRAINDIR/{}
+      if [ $n_test -gt $max_test ]
+      then
+        ls -rt $TESTDIR | head -n $overhead_test | xargs -I{} rm -f $TESTDIR/{}
+        let "n -= $overhead_test"
+        let "n_test -= $overhead_test"
+        echo -n "-"
+      fi
+      if [ $n_train -gt $max_train ]
+      then
+        ls -rt $TRAINDIR | head -n $overhead_train | xargs -I{} rm -f $TRAINDIR/{}
+        let "n -= $overhead_train"
+        let "n_train -= $overhead_train"
```

Comment:
This style of `let` is inconsistent with the ones at the top, is that intended (target variable inside/outside the quotes)? Do they operate differently?

Reply:
Its because its a mutating assignment operator rather than just a straight assignment.

Reply:
So this code ended up being wrong after all - it was inside a subshell so the variable gets updated correctly, but only in the subshell.  The test machine is running a fixed version.